core/src/serve_repair.rs

```diff
@@ -769,25 +710,16 @@ impl ServeRepair {
                 continue;
             }
 
-            let require_signature_check =
-                Self::requires_signature_check(&request, root_bank, sign_repairs_epoch);
-            if require_signature_check && !request.supports_signature() {
-                stats.err_unsigned += 1;
-                continue;
-            }
             if request.supports_signature()
                 && !Self::verify_signed_packet(&my_id, packet, &request, stats)
             {
                 continue;
             }
 
             let from_addr = packet.meta.socket_addr();
-            if let Some(ping_to_pkt) = Self::ping_to_packet_mapper_by_request_variant(
-                &request,
-                from_addr,
-                root_bank,
-                check_ping_ancestor_request_epoch,
-            ) {
+            if let Some(ping_to_pkt) =
+                Self::ping_to_packet_mapper_by_request_variant(&request, from_addr)
+            {
```

Comment:
The feature was never activated for mainnet so this change shouldn't affect mainnet behavior.

On testnet:
```
solana -u t feature status
...
Software Version         Feature Set  Stake   RPC   
1.15.0                     447902792   0.00%  0.09%  
1.14.3                    1060500786   0.14%  0.00%  
1.14.2                    1844441900  97.18% 94.48%  
1.11.10, 1.14.1, 1.14.2   4253057308   1.64%  5.18%  
1.11.6                    1113831037   0.00%  0.09%  
1.11.5                    3605932003   0.00%  0.17%  
```

Reply:
Here https://github.com/solana-labs/solana/pull/27910#issuecomment-1252487958:
 
> * if all clusters can ingest and process ping packets then the server should ping anyways regardless of requests being signed or not. No need to couple these two together.
> * if there are clusters which cannot ingest ping packets yet then the part sending ping packets needs to be entirely removed for now.

mainnet cannot process ping packets correctly because it does not have https://github.com/solana-labs/solana/pull/26878.
So the code should not send ping packets.
Also the code is tying ping packets to the signed requests (the `ping_to_packet_mapper_by_request_variant` function), which I don't think it is a good idea; easier if ping/pong is kept orthogonal to requests being signed or not.

Now because this code is tying ping packets to signed requests, this might be ok for mainnet since mainnet does not send signed requests yet. But given that there are multiple versions involved I would rather not to add this interdependency and additional complexity. Again, better if ping/pong is kept orthogonal to signing requests and does not depend on that.

So, I think it would be safest/easiest if we don't send any ping packets for now until all clusters can correctly ingest and process ping packets both on:
* repair socket which receives repair shreds.
* where it receives ancestor hash responses.